src/Swarrot/Processor/XDeath/XDeathMaxCountProcessor.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+namespace Swarrot\Processor\XDeath;
+
+use PhpAmqpLib\Wire\AMQPArray;
+use Psr\Log\LoggerInterface;
+use Psr\Log\LogLevel;
+use Psr\Log\NullLogger;
+use Swarrot\Broker\Message;
+use Swarrot\Broker\MessageProvider\MessageProviderInterface;
+use Swarrot\Processor\ProcessorInterface;
+use Swarrot\Processor\ConfigurableInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class XDeathMaxCountProcessor implements ConfigurableInterface
+{
+    /**
+     * @var ProcessorInterface
+     */
+    protected $processor;
+
+    /**
+     * @var MessageProviderInterface
+     */
+    private $messageProvider;
+
+    /**
+     * @var \Closure
+     */
+    private $callback;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @param ProcessorInterface $processor
+     * @param MessageProviderInterface $messageProvider
+     * @param \Closure $callback
+     * @param LoggerInterface|null $logger
+     */
+    public function __construct(
+        ProcessorInterface $processor,
+        MessageProviderInterface $messageProvider,
+        \Closure $callback,
+        LoggerInterface $logger = null
+    )
+    {
+        $this->processor = $processor;
+        $this->messageProvider = $messageProvider;
+        $this->callback = $callback;
+        $this->logger = $logger ?: new NullLogger();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setDefaultOptions(OptionsResolver $resolver)
+    {
+        $resolver
+            ->setDefaults(array(
+                'x_death_max_count' => 300,
+                'x_death_max_count_log_levels_map' => array(),
+                'x_death_max_count_fail_log_levels_map' => array(),
+            ))
+            ->setAllowedTypes('x_death_max_count', 'int')
+            ->setAllowedTypes('x_death_max_count_log_levels_map', 'array')
+            ->setAllowedTypes('x_death_max_count_fail_log_levels_map', 'array');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function process(Message $message, array $options)
+    {
+        try {
+            return $this->processor->process($message, $options);
+        } catch (\Throwable $e) {
+            return $this->handleException($e, $message, $options);
+        } catch (\Exception $e) {
+            return $this->handleException($e, $message, $options);
+        }
+    }
+
+    /**
+     * @param $exception
+     * @param Message $message
+     * @param array $options
+     *
+     * @return mixed
+     */
+    private function handleException($exception, Message $message, array $options)
+    {
+        $headers = $message->getProperties();
+        if (isset($headers['headers']['x-death'], $options['x_death_max_count'])) {
```

Comment:
no need to check whether `$options['x_death_max_count']` is set. It will always be set when using the processor in a supported way, as a default value is defined for the option